code/modules/flufftext/Hallucination.dm

```diff
@@ -315,15 +357,24 @@ GLOBAL_LIST_INIT(hallucination_list, list(
 		target.client.images |= fakerune
 	target.playsound_local(wall,'sound/effects/meteorimpact.ogg', 150, 1)
 	bubblegum = new(wall, target)
-	addtimer(CALLBACK(src, .proc/bubble_attack, landing), 10)
+	addtimer(CALLBACK(src, .proc/start_processing), 10)
 
-/datum/hallucination/oh_yeah/proc/bubble_attack(turf/landing)
-	var/charged = FALSE //only get hit once
-	while(get_turf(bubblegum) != landing && target?.stat != DEAD)
-		if(!landing)
-			break
-		if((get_turf(bubblegum)).loc.z != landing.loc.z)
-			break
+/datum/hallucination/oh_yeah/proc/start_processing()
+	if (isnull(target))
+		qdel(src)
+		return
+	START_PROCESSING(SSfastprocess, src)
+
+/datum/hallucination/oh_yeah/process(delta_time)
+	time_spent += delta_time
+
+	if (time_spent < 0.2)
+		return
```

Comment:
Doesn't mean that the initial attack will be delayed by 0.2 seconds?

Reply:
Yeah, I'll fix it.